core/include/detray/core/detail/context_aware_store.hpp

```diff
@@ -0,0 +1,360 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s)
+#include "detray/core/detail/container_buffers.hpp"
+#include "detray/core/detail/container_views.hpp"
+#include "detray/core/detail/data_context.hpp"
+#include "detray/definitions/detail/indexing.hpp"
+#include "detray/definitions/detail/qualifiers.hpp"
+
+// Vecmem include(s)
+#include <vecmem/memory/memory_resource.hpp>
+
+// System include(s)
+#include <type_traits>
+
+namespace detray {
+
+/// @brief Wraps a jagged vector-like container and implements a data store
+/// around it.
+///
+/// @tparam T The type of the collection data, e.g. transforms
+/// @tparam container_t The type of container to use for the data collection.
+/// @tparam context_t the context with which to retrieve the correct data.
+template <typename T, template <typename...> class container_t,
+          typename context_t>
+class context_aware_store {
+    public:
+    using base_type = container_t<T>;
+    using vector_type = typename base_type::value_type;
+    using context_type = context_t;
+
+    // TODO: do we need this?
```

Comment:
These are used by the detector to figure which type of indexed based link to use with this store. For transforms in AoS layout that will be a ```single_link```